src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Controllers/AuthenticationController.cs

```diff
@@ -340,8 +343,58 @@ private async Task<ActionResult> AuthenticateMaskinportenToken(string originalTo
                     }
                 }
 
+                string authenticatemethod = "maskinporten";
+
+                if (!string.IsNullOrEmpty(Request.Headers["X-Altinn-EnterpriseUser-Authentication"]))
+                {
+                    string enterpriseUserHeader = Request.Headers["X-Altinn-EnterpriseUser-Authentication"];
+                    byte[] decodedCredentials = Convert.FromBase64String(enterpriseUserHeader);
```

Comment:
I'm assuming this will throw if a invalid base64 encoded string is supplied. Should probably extract the parsing of this header to a helper, where exceptions are caught and null checks are performed. As per the acceptance criteria, invalid data here should result in a 400 Bad request.

Reply:
Moved decoding of Base64 to a helper-method that returns a 400 if an exception is caught.